### PR TITLE
[MISC] Update Juju 3.4 pin to Juju 3.5

### PR DESCRIPTION
**Pull request by dragomirp**
~~Run all Juju 3 integration tests on 3.5~~
* Move Juju 3.4 pin to 3.5
* Enable Mongo K8s tests to run on 3.5

### CODE REVIEW

File: tests/integration/test_mongodb.py [hunk 1]
```diff
@@ -55,16 +56,10 @@ async def test_deploy(
...
-        "3.4."
-    ) and cloud_name == "microk8s":
-        pytest.skip("Test is incompatible with Juju 3.4 on microk8s")
-
-    channel = "5/edge" if cloud_name == "localhost" else "edge"
     await asyncio.gather(
         ops_test.model.deploy(
             MONGODB[cloud_name],
-            channel=channel,
+            channel="6/edge",
```
Comment (dragomirp) [inline tests/integration/test_mongodb.py:62]: Both k8s and VM have a 6/edge branch now

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -69,8 +69,7 @@ jobs:
         juju:
           - agent: 2.9.50  # renovate: juju-agent-pin-minor
             libjuju: ^2
-          - agent: 3.1.9  # renovate: juju-agent-pin-minor
-          - agent: 3.4.5  # renovate: juju-agent-pin-minor
+          - agent: 3.5.3  # renovate: juju-agent-pin-minor
```
Comment (dragomirp) [inline .github/workflows/ci.yaml:72]: One run on Juju 3 on 3.5

File: tests/integration/test_mongodb.py (hunk 1 above)
Comment (taurus-forever) [reply tests/integration/test_mongodb.py:62]: Uff, we were testing here deprecated 5/edge. cc: @Mehdi-Bendriss

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (taurus-forever) [reply .github/workflows/ci.yaml:72]: сс: @delgod are you OK?  It is a shared component....

**Review by taurus-forever — APPROVED** [review]

File: tests/integration/test_mongodb.py (hunk 1 above)
Comment (dragomirp) [reply tests/integration/test_mongodb.py:62]: I think juju 3.1 was installing the correct revision https://github.com/canonical/data-integrator/actions/runs/10875992483/job/30175392118#step:31:114

So we didn't see that the channel was wrong.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (dragomirp) [reply .github/workflows/ci.yaml]: Leaving tests to 2.9, 3.1 and 3.5 for now, on @delgod's request.

**Review by Mehdi-Bendriss — APPROVED** [review]

**Pull request merged**